src/Configuration/ConfigurationSet.php

```diff
@@ -32,7 +33,8 @@ protected function assertValid(array $classes)
         parent::assertValid($classes);
 
         foreach ($classes as $config) {
-            if (!is_subclass_of($config, ConfigurationInterface::class)) {
+            $reflector = new ReflectionClass($config);
+            if (!$reflector->implementsInterface(ConfigurationInterface::class)) {
```

Comment:
The bug in question was reported [against 5.3.5](https://bugs.php.net/bug.php?id=53727) and has been fixed since 2012.